src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -113,6 +113,11 @@ public void deleteCard(Card target) {
     @Override
     public void addCard(Card card) {
         masterDeck.addCard(card);
+
+        // only update filteredCardList if a deck is selected
+        Optional.ofNullable(selectedDeck)
```

Comment:
I think this is not required anymore because filteredCardList is an instance of an ObservableList which listens to master deck. @unfazing do correct me if I am wrong! Smt we found out today.